trailblazer/clients/authentication_client/google_oauth_client.py

```diff
@@ -0,0 +1,31 @@
+import requests
+
+from trailblazer.clients.authentication_client.dtos.tokens_request import GetTokensRequest
+from trailblazer.clients.authentication_client.dtos.tokens_response import TokensResponse
+from trailblazer.clients.authentication_client.exceptions import GoogleOAuthClientError
+
+
+class GoogleOAuthClient:
+
+    def __init__(self, client_id: str, client_secret: str, redirect_uri: str, token_uri: str):
+        self.client_id = client_id
+        self.client_secret = client_secret
+        self.token_uri = token_uri
+        self.redirect_uri = redirect_uri
+
+    def get_tokens(self, authorization_code: str) -> TokensResponse:
+        """Exchange the authorization code for an access token and refresh token."""
+        request = GetTokensRequest(
+            client_id=self.client_id,
+            client_secret=self.client_secret,
+            code=authorization_code,
+            redirect_uri=self.redirect_uri,
+        )
+        data: str = request.model_dump_json()
+
+        response = requests.post(self.token_uri, data=data)
```

Comment:
You are right, changed to `model_dump`

Reply:
It might be better with `model_dump_json` though, I think it supports serializing date time objects for example. But then you would need to set it as the json parameter in the requests.post. I.e. `requests.post(self.token_uri, json=json_dump)`. Might need more work though, haven't used it too much.

Reply:
Google uses `data=...` in their example, so I'll stick to it for now!

Reply:
https://github.com/Clinical-Genomics/trailblazer/pull/402/commits/8f577c23e578b7ac434d949795718a0d6808a4ba